dev-libs/jemalloc/jemalloc-5.0.1.recipe

```diff
@@ -0,0 +1,90 @@
+SUMMARY="A general-purpose scalable concurrent allocator"
+DESCRIPTION="jemalloc is a general purpose malloc(3) implementation that \
+emphasizes fragmentation avoidance and scalable concurrency support. \
+jemalloc first came into use as the FreeBSD libc allocator in 2005, and since \
+then it has found its way into numerous applications that rely on its \
+predictable behavior. In 2010 jemalloc development efforts broadened to \
+include developer support features such as heap profiling and extensive \
+monitoring/tuning hooks. Modern jemalloc releases continue to be integrated \
+back into FreeBSD, and therefore versatility remains critical. Ongoing \
+development efforts trend toward making jemalloc among the best allocators \
+for a broad range of demanding applications, and eliminating/mitigating \
+weaknesses that have practical repercussions for real world applications."
+HOMEPAGE="https://jemalloc.net/"
+COPYRIGHT="2002-2017 Jason Evans
+	2007-2012 Mozilla Foundation
+	2009-2017 Facebook, Inc."
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/jemalloc/jemalloc/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="5de6dcb50de04b14bae9340d23515390925fcaa7637341707239d09a3ea07a23"
+PATCHES="jemalloc-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	jemalloc$secondaryArchSuffix = $portVersion compat >= 5
+	cmd:jemalloc.sh
+	cmd:jeprof
+	lib:libjemalloc$secondaryArchSuffix = 2 compat >= 2
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	jemalloc${secondaryArchSuffix}_devel = $portVersion compat >= 5
+	cmd:jemalloc_config
+	devel:libjemalloc$secondaryArchSuffix = 2 compat >= 2
+	devel:libjemalloc_pic$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	jemalloc$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:automake
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:xsltproc #required
+	"
+
+defineDebugInfoPackage jemalloc$secondaryArchSuffix \
+	$libDir/libjemalloc.so.2
+
+BUILD()
+{
+	autoconf
+	runConfigure ./configure \
+		--enable-autogen
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install_bin install_include install_lib
+
+	prepareInstalledDevelLibs libjemalloc libjemalloc_pic
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel \
+		$developDir \
+		$prefix/bin/jemalloc-config
+}
+
+TEST()
+{
+	make check
```

Comment:
I think you will need to pass a LIBRARY_PATH here, probably something like:
```diff
 TEST()
 {
+	LIBRARY_PATH="$sourceDir/lib${LIBRARY_PATH:+:$LIBRARY_PATH}" \
 	make check
 }
```

Reply:
It works already great, no need for more patching here.